mobile/native/src/unfunded_orders.rs

```diff
@@ -0,0 +1,123 @@
+use crate::event;
+use crate::event::EventInternal;
+use crate::event::FundingChannelTask;
+use crate::state;
+use crate::trade::order;
+use crate::trade::order::api::NewOrder;
+use anyhow::Result;
+use bitcoin::Address;
+use bitcoin::Amount;
+use futures::FutureExt;
+use std::time::Duration;
+use xxi_node::commons::ChannelOpeningParams;
+
+pub(crate) async fn submit_unfunded_channel_opening_order(
+    funding_address: Address,
+    new_order: NewOrder,
+    coordinator_reserve: u64,
+    trader_reserve: u64,
+    needed_channel_size: u64,
+) -> Result<()> {
+    let node = state::get_node().clone();
+    let bdk_node = node.inner.clone();
+    event::publish(&EventInternal::FundingChannelNotification(
+        FundingChannelTask::Pending,
+    ));
+    let runtime = crate::state::get_or_create_tokio_runtime()?;
+    let (future, remote_handle) = async move {
+        event::publish(&EventInternal::FundingChannelNotification(
+            FundingChannelTask::Pending,
+        ));
```

Comment:
❓ Why do you need to send `Pending` so many times?

Reply:
good question 🙈